### PR TITLE
Improve `LIBASSERT_VA_ARGS` macro

### PR DESCRIPTION
**Pull request by Eczbek**
Also use in `internal_verify` macro

### CODE REVIEW

File: include/libassert/assert.hpp [hunk 1]
```diff
@@ -137,6 +137,12 @@
  #pragma warning(disable: 4251; disable: 4275)
 #endif
 
+#if __cplusplus >= 2020002
+ #define LIBASSERT_VA_ARGS(...) __VA_OPT__(,) __VA_ARGS__
+#else
```
Comment (jeremy-rifkin) [inline include/libassert/assert.hpp:142]: Can you update this to `LIBASSERT_IS_GCC || __cplusplus >= 2020002`. The library still needs to workaround https://gcc.gnu.org/bugzilla/show_bug.cgi?id=44317 (and if you could re-add the comment that'd be great).

**Comment by jeremy-rifkin** [comment]
The sonar failure is ok, that's not related to your PR

File: include/libassert/assert.hpp (hunk 1 above)
Comment (Eczbek) [reply include/libassert/assert.hpp:142]: sure

**Review by jeremy-rifkin — APPROVED** [review]
I think looks good to me

**Pull request merged**